src/node_http_parser.cc

```diff
@@ -442,14 +434,8 @@ class Parser : public AsyncWrap, public StreamListener {
 
     argv[A_UPGRADE] = Boolean::New(env()->isolate(), parser_.upgrade);
 
-    MaybeLocal<Value> head_response;
-    {
-      InternalCallbackScope callback_scope(
-          this, InternalCallbackScope::kSkipTaskQueues);
-      head_response = cb.As<Function>()->Call(
-          env()->context(), object(), arraysize(argv), argv);
-      if (head_response.IsEmpty()) callback_scope.MarkAsFailed();
-    }
+    MaybeLocal<Value> head_response = cb.As<Function>()->Call(
+        env()->context(), object(), arraysize(argv), argv);
```

Comment:
Likewise here, this is not properly handling errors.

Reply:
We don't need to handle the errors since this is the callback provided by the user. In this case, we're the user. And if callback fails, we don't need to trigger an async_hook callback to report the error.